### PR TITLE
pulse: convert to cffi's in-line API mode

### PR DESCRIPTION
**Pull request by tych0**
This still doesn't work, I get:

>>> from libqtile.widget import PulseVolume
Traceback (most recent call last):
  File "/home/tycho/.local/lib/python3.11/site-packages/cffi/api.py", line 183, in _typeof
    result = self._parsed_types[cdecl]
             ~~~~~~~~~~~~~~~~~~^^^^^^^
KeyError: 'void(pa_context *c, const pa_sink_info *i, int eol, void *userdata)'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/tycho/.local/lib/python3.11/site-packages/cffi/api.py", line 437, in _get_cached_btype
    BType = self._cached_btypes[type]
            ~~~~~~~~~~~~~~~~~~~^^^^^^
KeyError: <void(*)(pa_context *, pa_sink_info const *, int, void *)>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/tycho/.local/lib/python3.11/site-packages/cffi/model.py", line 58, in get_cached_btype
    BType = ffi._cached_btypes[self]
            ~~~~~~~~~~~~~~~~~~^^^^^^
KeyError: <void(*)(pa_context *, pa_sink_info const *, int, void *)>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/tycho/.local/lib/python3.11/site-packages/cffi/model.py", line 58, in get_cached_btype
    BType = ffi._cached_btypes[self]
            ~~~~~~~~~~~~~~~~~~^^^^^^
KeyError: <pa_sink_info const *>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/tycho/.local/lib/python3.11/site-packages/cffi/model.py", line 58, in get_cached_btype
    BType = ffi._cached_btypes[self]
            ~~~~~~~~~~~~~~~~~~^^^^^^
KeyError: <pa_sink_info>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<frozen importlib._bootstrap>", line 1231, in _handle_fromlist
  File "/home/tycho/.local/lib/python3.11/site-packages/libqtile/utils.py", line 247, in __getattr__
    return import_class(module_path, name, fallback=fallback)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/tycho/.local/lib/python3.11/site-packages/libqtile/utils.py", line 218, in import_class
    module = importlib.import_module(module_path, __package__)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/tycho/.local/lib/python3.11/site-packages/libqtile/widget/pulse_volume.py", line 18, in <module>
    @ffi.callback("void(pa_context *c, const pa_sink_info *i, int eol, void *userdata)")
     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/tycho/.local/lib/python3.11/site-packages/cffi/api.py", line 399, in callback
    cdecl = self._typeof(cdecl, consider_function_as_funcptr=True)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/tycho/.local/lib/python3.11/site-packages/cffi/api.py", line 186, in _typeof
    result = self._typeof_locked(cdecl)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/tycho/.local/lib/python3.11/site-packages/cffi/api.py", line 175, in _typeof_locked
    btype = self._get_cached_btype(type)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/tycho/.local/lib/python3.11/site-packages/cffi/api.py", line 440, in _get_cached_btype
    BType = type.get_cached_btype(self, finishlist)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/tycho/.local/lib/python3.11/site-packages/cffi/model.py", line 60, in get_cached_btype
    BType = self.build_backend_type(ffi, finishlist)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/tycho/.local/lib/python3.11/site-packages/cffi/model.py", line 246, in build_backend_type
    args.append(tp.get_cached_btype(ffi, finishlist))
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/tycho/.local/lib/python3.11/site-packages/cffi/model.py", line 60, in get_cached_btype
    BType = self.build_backend_type(ffi, finishlist)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/tycho/.local/lib/python3.11/site-packages/cffi/model.py", line 273, in build_backend_type
    BItem = self.totype.get_cached_btype(ffi, finishlist, can_delay=True)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/tycho/.local/lib/python3.11/site-packages/cffi/model.py", line 397, in get_cached_btype
    BType = StructOrUnionOrEnum.get_cached_btype(self, ffi, finishlist,
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/tycho/.local/lib/python3.11/site-packages/cffi/model.py", line 60, in get_cached_btype
    BType = self.build_backend_type(ffi, finishlist)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/tycho/.local/lib/python3.11/site-packages/cffi/model.py", line 474, in build_backend_type
    self.check_not_partial()
  File "/home/tycho/.local/lib/python3.11/site-packages/cffi/model.py", line 471, in check_not_partial
    raise VerificationMissing(self._get_c_name())
cffi.VerificationMissing: pa_sink_info

when importing it. but hopefully it's close?

### CODE REVIEW

**Comment by tych0** [comment]
I'm pretty sure this won't work. I played around with it for a few hours today, and our combination of callback usage + enum usage etc. makes this switch quite complicated. I've got one more idea that I don't have time to try right now (actually expand all the enums etc.), but even that is quite a hack and maybe we shouldn't merge it.

tldr: I think we need a release plan where we don't use in-line mode for pulse.
